src/core/meshcop/dataset_manager.cpp

```diff
@@ -246,12 +246,25 @@ void DatasetManager::HandleTimer(void *aContext)
 
 void DatasetManager::HandleTimer(void)
 {
+    const Timestamp *localActiveTimestamp = mNetif.GetActiveDataset().GetLocal().GetTimestamp(Tlv::kActiveTimestamp);
+    const Timestamp *pendingActiveTimestamp = mNetif.GetPendingDataset().GetNetwork().GetTimestamp(Tlv::kActiveTimestamp);
```

Comment:
You could avoid most of the changes here by using `Dataset::Set()` here instead to grab `Tlv::kActiveTimestamp`.  Thoughts?